### ISSUE TITLE
Test on Eclipse Photon

### ISSUE THREAD

**Issue by TDesjardins** [gwt-plugins/gwt-eclipse-plugin#380]
It's time to test the gwt-eclipse-plugin with Eclipse Photon because of the upcoming release in June. Currently M6 is available for download:
http://www.eclipse.org/downloads/packages/release/Photon/M6

**Comment by branflake2267** [comment]
Sorry for the delay. I've been thinking about this too. I'm aiming to schedule time soon on working on Eclipse as well.

**Comment by TDesjardins** [comment]
I did some testing on Photon RC3 (GWT compile, superdev, GWT project creation). Seems that the GWT plugin works fine on Photon.

**Comment by CrushaKRool** [comment]
While the Oxygen release of the plugin seems to work fine with Photon, I did encounter some compile issues due to changed API when opening the project in Photon. I've outlined the changes here:
fb7971883eeac0f9385297bddae0875f2a768c74

For the most part, it seems to be about changes to the internal API that is used in the project in some places. I wonder if that could cause problems under specific circumstances during runtime.


Are there plans to keep supporting Eclipse releases all the way back to Mars in the same code base, or is it conceivable to create a maintenance branch for older releases that get deployed separately and make way for newer releases to use the corresponding newer Eclipse API?
